install/test/helm_test.go

```
@@ -98,15 +96,21 @@ var _ = Describe("Helm Test", func() {
 			statsAnnotations map[string]string
 		)
 
-		expectCustomResourceSubstring := func(substr string, contains bool) {
+		// Create a TestManifest out of the custom resource yaml from the configmap,
+		// and then perform assertions on the resources (which are assumed to all be Gateways).
+		// `matchers` is a map of Gateway name to the assertion to be made on that Gateway.
+		// There must be a matcher for every Gateway that's expected to be in the manifest.
```

From reading the comment it seems like the test should fail if there is an unexpected gateway in the manifest, but it looks like the code will simply not make an assertion on any gateway for which there is not matcher

I'm having issues running the tests locally which are unrelated, so I'm unable to explore this independently, but assuming this is the case we should either clarify the comment or modify the code to reflect the intended behavior

---

below it checks that the number of resources found in the manifest matches the number of matchers you passed in:
`ExpectWithOffset(1, customResourceManifest.NumResources()).To(Equal(len(matchers)))`
so we shouldn't have a case where there is an extra gateway without an associated matcher.
i can add a comment on that line so it's more apparent

---

Ah, yup, I missed that
As long as it does what the comment says I think it's good as-is, I was just unable to verify myself